single_kernel_mongo/events/backups.py

```diff
@@ -122,6 +123,15 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent) -> None:  #
             )
             return
 
+        match self.manager.backup_state():
+            case BackupState.BACKUP_RUNNING | BackupState.RESTORE_RUNNING:
```

Comment:
I'm not sure this can work in this case, if the credentials have changed, there is no guarantee the previous credentials are still valid for the backup / restore to continue

Reply:
The credentials haven't been rotated yet, so it still uses the former credentials until they finish rotating.

Reply:
I mean the latter, the credentials may no longer be accepted by the cloud